homeassistant/config.py

```diff
@@ -1025,21 +1046,167 @@ async def merge_packages_config(
+
+    In case of errors:
+    - Print the error messages to the log.
+    - Raise a ConfigValidationError if raise_on_failure is set.
+
+    Returns the integration config or `None`.
+    """
+    integration_config_info = await async_process_component_config(
+        hass, config, integration
+    )
+    return async_handle_component_errors(
+        hass, integration_config_info, integration, raise_on_failure
+    )
 
-    Returns None on error.
+
+@callback
+def async_handle_component_errors(
+    hass: HomeAssistant,
+    integration_config_info: IntegrationConfigInfo,
+    integration: Integration,
+    raise_on_failure: bool = False,
+) -> ConfigType | None:
+    """Handle component configuration errors from async_process_component_config.
+
+    In case of errors:
+    - Print the error messages to the log.
+    - Raise a ConfigValidationError if raise_on_failure is set.
+
+    Returns the integration config or `None`.
+    """
+
+    processed_config = integration_config_info.config
+    config_exception_info = integration_config_info.exception_info_list
+    if not config_exception_info:
+        return processed_config
+
+    platform_exception: ConfigExceptionInfo
+    domain = integration.domain
+    placeholders: dict[str, str] = {
+        "domain": domain,
+    }
+    for platform_exception in config_exception_info:
+        exception = platform_exception.exception
+        placeholders["error"] = str(exception)
+        platform_name = placeholders["p_name"] = platform_exception.platform_name
+        platform_config = platform_exception.config
+        log_message, show_stack_trace = _get_log_message_and_stack_print_pref(
+            platform_exception.translation_key, exception, domain, platform_name
+        )
+        link = platform_exception.integration_link or integration.documentation
+        if log_message is None:
+            # If no pre defined log_message is set, we generate an enriched error
+            # message, so we can notify about it during setup
+            if isinstance(exception, vol.Invalid):
+                log_message = format_schema_error(
+                    hass, exception, platform_name, platform_config, link
+                )
+                if annotation := find_annotation(platform_config, exception.path):
+                    placeholders["config_file"], line = annotation
+                    placeholders["line"] = str(line)
+            else:
+                if TYPE_CHECKING:
+                    assert isinstance(exception, HomeAssistantError)
+                log_message = format_homeassistant_error(
+                    hass, exception, platform_name, platform_config, link
+                )
+                if annotation := find_annotation(platform_config, [platform_name]):
+                    placeholders["config_file"], line = annotation
+                    placeholders["line"] = str(line)
+                show_stack_trace = True
```

Comment:
I think this can be moved to inside of `_get_log_message_and_stack_print_pref` now